thunder/executors/torch_autograd.py

```diff
@@ -325,6 +325,13 @@ def make_trace(func):
     if getattr(compile_data.fn, "use_ddp", False):
         bw_extrace = sort_waits(bw_extrace)
 
+    # Importing here to avoid cyclical dependencies in future.
+    from thunder.executors.transformer_engineex import transformer_engine_ex, _rearrange_transformer_engine_linear
+
+    if transformer_engine_ex in compile_data.executors_list:
+        # NOTE: `_rearrange_transformer_engine_linear` mutates `fw_extrace`.
+        _rearrange_transformer_engine_linear(fw_extrace, bw_extrace)
```

Comment:
Maybe in the future TransformerEngine should become a transform that automatically enables the operator to avoid having this hardcoded here.

cc @t-vi since we've talked about this and it's similar to fsdp